sdk/core/core/src/az_span.c

```diff
@@ -618,3 +618,88 @@ _az_span_scan_until(az_span self, _az_predicate predicate, int32_t* out_index)
   }
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
+
+AZ_NODISCARD az_span _az_span_trim_white_space(az_span source)
+{
+  // az_span source with size 0 is allowed. For instance, a header value can be AZ_SPAN_NULL
+  if (az_span_size(source) == 0)
```

Comment:
nit: How about something like this, to avoid any extra checks that won't be true in most cases anyway?

With the clamping helper methods, the implementation of trim becomes quite straightforward. Just a couple of lines each and you avoid some code duplication.

```C
AZ_NODISCARD int32_t _az_span_clamp_start(az_span source)
{
  int32_t start = 0;
  
  uint8_t* source_ptr = az_span_ptr(source);

  for (; start < az_span_size(source); start++)
  {
    if (!_az_is_white_space(source_ptr[start]))
    {
      break;
    }
  }

  return start;
}

AZ_NODISCARD int32_t _az_span_clamp_end(az_span source, int32_t start)
{
  // If _az_span_clamp_start trims all, then start == size
  AZ_PRECONDITION(start >= 0 && start <= az_span_size(source));

  int32_t end = az_span_size(source) - 1;

  uint8_t* source_ptr = az_span_ptr(source);

  for (; end >= start; end--)
  {
    if (!_az_is_white_space(source_ptr[end]))
    {
      break;
    }
  }

  return end;

}

AZ_NODISCARD az_span _az_span_trim_white_space(az_span source)
{
  int start = _az_span_clamp_start(source);
  int end = _az_span_clamp_end(source, start);
  return az_span_slice(source, start, end);
}

AZ_NODISCARD az_span _az_span_trim_white_space_from_start(az_span source)
{
   return az_span_slice_to_end(source, _az_span_clamp_start(source));
}

AZ_NODISCARD az_span _az_span_trim_white_space_from_end(az_span source)
{
   return az_span_slice(source, 0, _az_span_clamp_end(source, 0));
}
```

Reply:
it's the same code like.  You still need two extra functions to be called from ...start() and ...end().
I don't see a real difference.
I though about a more general trim method that receives a `reverse` parameter to start from the end, but it ends up checking that parameter everytime inside the loop, so, I decide leaving it simple without more functions